src/EnergyPlus/CommandLineInterface.cc

```diff
@@ -95,163 +94,183 @@ namespace CommandLineInterface {
 
-        opt.get("-j")->getInt(state.dataGlobal->numThread);
+        app.add_option("-d,--output-directory", state.dataStrGlobals->outDirPath, "Output directory path (default: current directory)")
+            ->option_text("DIR")
+            ->required(false);
+        // ->check(CLI::ExistingDirectory) // We don't require it to exist, we make it if needed
 
-        if (state.dataGlobal->numThread == 0) {
-            DisplayString(state, "Invalid value for -j arg. Defaulting to 1.");
-            state.dataGlobal->numThread = 1;
-        } else if (state.dataGlobal->numThread > (int)std::thread::hardware_concurrency()) {
-            DisplayString(state,
-                          fmt::format("Invalid value for -j arg. Value exceeds num available. Defaulting to num available. -j {}",
-                                      (int)std::thread::hardware_concurrency()));
-            state.dataGlobal->numThread = (int)std::thread::hardware_concurrency();
-        }
-
-        // Process standard arguments
-        if (opt.isSet("-h")) {
-            DisplayString(state, usage);
+        if (legacyMode) {
+            state.dataStrGlobals->inputIddFilePath = "Energy+.idd";
+        } else {
+            state.dataStrGlobals->inputIddFilePath = state.dataStrGlobals->exeDirectoryPath / "Energy+.idd";
+        }
+        app.add_option(
+               "-i,--idd", state.dataStrGlobals->inputIddFilePath, "Input data dictionary path (default: Energy+.idd in executable directory)")
+            ->required(false)
+            ->option_text("IDD")
+            ->check(CLI::ExistingFile);
+
+        bool runEPMacro = false;
+        app.add_flag("-m,--epmacro", runEPMacro, "Run EPMacro prior to simulation");
+
+        std::string prefixOutName = "eplus";
+        app.add_option("-p,--output-prefix", prefixOutName, "Prefix for output file names (default: eplus)")->required(false)->option_text("PRE");
+
+        app.add_flag("-r,--readvars", state.dataGlobal->runReadVars, "Run ReadVarsESO after simulation");
+
+        app.add_flag("-c,--convert", state.dataGlobal->outputEpJSONConversion, "Output IDF->epJSON or epJSON->IDF, dependent on input file type");
+
+        app.add_flag("--convert-only",
+                     state.dataGlobal->outputEpJSONConversionOnly,
+                     "Only convert IDF->epJSON or epJSON->IDF, dependent on input file type. No simulation");
+
+        // TODO: make it an enum instead?
+        std::string suffixType = "L";
+        std::string const suffixHelp = R"help(Suffix style for output file names (default: L)
+   L: Legacy (e.g., eplustbl.csv)
+   C: Capital (e.g., eplusTable.csv)
+   D: Dash (e.g., eplus-table.csv))help";
+        app.add_option("-s,--output-suffix", suffixType, suffixHelp)
+            ->option_text("SUFFIX")
+            ->required(false)
+            ->check(CLI::IsMember({"L", "C", "D"}, CLI::ignore_case));
```

Comment:
I can see what's going on here.  It's a bit funny looking, and I would have a tough time just rattling off this code from memory.  But it's the CLI...it's not going to change almost ever.  And if it does, you've given plenty of examples we can base changes off of.  This is good.

Reply:
Is this specific to this option or generally speaking?

Reply:
Generally speaking.  The chain of trailing method calls honestly looks like..... Javascript.  `.then(function() { .then(function() ....`  It's totally fine and I'm not saying anything is wrong.  Just noting.